python/ray/worker.py

```diff
@@ -2457,19 +2515,45 @@ def remote(*args, **kwargs):
     """
     worker = global_worker
 
-    def make_remote_decorator(num_return_vals, resources, max_calls,
-                              checkpoint_interval, func_id=None):
+    def make_remote_decorator(num_return_vals, num_cpus, num_gpus, resources,
+                              max_calls, checkpoint_interval, func_id=None):
         def remote_decorator(func_or_class):
             if inspect.isfunction(func_or_class) or is_cython(func_or_class):
+                # Set the remote function default resources.
+                resources["CPU"] = (DEFAULT_REMOTE_FUNCTION_CPUS
```

Comment:
so if someone does `num_gpus=2, custom_resources={"GPU": 1}` custom GPUs will be overwritten?

Reply:
If you include `"GPU"` in `resources=` then it will raise an exception. That can only be specified through `num_gpus=`.